validations/src/validations.rs

```diff
@@ -418,6 +395,41 @@ pub fn validate_consensus(
     }
 }
 
+/// Construct a `RadonReport` from a `TallyPreconditionClauseResult`
+pub fn construct_report_from_clause_result(
+    clause_result: Result<TallyPreconditionClauseResult, RadError>,
+    script: &RADTally,
+    reports_len: usize,
+) -> Result<RadonReport<RadonTypes>, RadError> {
+    match clause_result {
+        // The reveals passed the precondition clause (a parametric majority of them were successful
+        // values). Run the tally, which will add more liars if any.
+        Ok(TallyPreconditionClauseResult::MajorityOfValues { values, liars }) => {
+            run_tally_report(values, script, Some(liars))
+        }
+        // The reveals did not pass the precondition clause (a parametric majority of them were
+        // errors). Tally will not be run, and the mode of the errors will be committed.
+        Ok(TallyPreconditionClauseResult::MajorityOfErrors { errors_mode }) => {
+            // Do not impose penalties on any of the revealers.
+            let mut metadata = TallyMetaData::default();
+            metadata.update_liars(vec![false; reports_len]);
+
+            RadonReport::from_result(
+                Ok(RadonTypes::RadonError(errors_mode)),
+                &ReportContext::from_stage(Stage::Tally(metadata)),
+            )
+        }
+        // Failed to evaluate the precondition clause. `RadonReport::from_result()?` is the last
+        // chance for errors to be intercepted and used for consensus.
+        Err(e) => {
+            let mut metadata = TallyMetaData::default();
+            metadata.liars = vec![false; reports_len];
```

Comment:
This should use `metadata.update_liars` just like the previous case?

Reply:
True